p4-16/spec/P4-16-spec.mdk

```diff
@@ -3220,6 +3222,20 @@ type. The result always has the same width as the left operand.
   - all result bits are zero when shifting left
   - all result bits are zero when shifting right a positive value
   - all result bits are one when shifting right a negative value
+- Extraction of a set of contiguous bits, also known as a slice, denoted by `[m:l]`,
+  where `m` and `l` must be positive integers
+  that are compile-time known values, and `m >= l`. The result is
+  a bit-string of width `m - l + 1`, including the bits numbered
+  from `l` (which becomes the least significant bit of the result) to `m` (the
+  most significant bit of the result) from the source operand. The conditions `0 <= l < W`
+  and `l <= m < W` are checked statically (where `W` is
+  the length of the source bit-string). Note that both endpoints of
+  the extraction are inclusive. The bounds are required to be
+  compile-time known values so that the result width can be computed
+  at compile time. Slices are also l-values, which means that P4 supports assigning to a slice: ` e[m:l] = x `.
+  The effect of this statement is to set bits `m` to `l` of `e` to the
+  bit-pattern represented by `x`, and leaves all other bits of `e`
+  unchanged.  A slice of a signed integer is treated like a signed integer as well.
```

Comment:
Would it be reasonable to define a slice on type `int<W> e` to be equal to `(int<X>) ( ((bit<X>) e)[m:l] ), where `X=m-l+1` ?

Reply:
that's a way to define the value of this expression, but it does not work as an l-value, since casts are not l-values.

Reply:
Resolving this Github conversation, since Mihai answered it to my satisfaction.